pkg/asset/installconfig/gcp/validation.go

```diff
@@ -17,16 +17,69 @@ import (
 	"github.com/openshift/installer/pkg/types"
 )
 
+type resourceRequirements struct {
+	minimumVCpus  int64
+	minimumMemory int64
+}
+
+var controlPlaneReq = resourceRequirements{
+	minimumVCpus:  4,
+	minimumMemory: 16384,
+}
+
+var computeReq = resourceRequirements{
+	minimumVCpus:  2,
+	minimumMemory: 8192,
+}
+
 // Validate executes platform-specific validation.
 func Validate(client API, ic *types.InstallConfig) error {
 	allErrs := field.ErrorList{}
 
 	allErrs = append(allErrs, validateProject(client, ic, field.NewPath("platform").Child("gcp"))...)
 	allErrs = append(allErrs, validateNetworks(client, ic, field.NewPath("platform").Child("gcp"))...)
 
+	if ic.ControlPlane != nil && ic.ControlPlane.Platform.GCP != nil && ic.ControlPlane.Platform.GCP.InstanceType != "" {
+		allErrs = append(allErrs, validateInstanceType(client, ic, field.NewPath("controlPlane", "platform", "gcp"), ic.ControlPlane.Platform.GCP.InstanceType, controlPlaneReq)...)
+	}
+
+	for idx, compute := range ic.Compute {
+		fieldPath := field.NewPath("compute").Index(idx)
+		if compute.Platform.GCP != nil && compute.Platform.GCP.InstanceType != "" {
+			allErrs = append(allErrs, validateInstanceType(client, ic, fieldPath.Child("platform", "gcp"), compute.Platform.GCP.InstanceType, computeReq)...)
+		}
+	}
+
 	return allErrs.ToAggregate()
 }
 
+// ValidateInstanceType ensures the macyhine type has sufficient Vcpu and Memory
+func validateInstanceType(client API, ic *types.InstallConfig, fieldPath *field.Path, instanceType string, req resourceRequirements) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if instanceType != "" {
+		filter := fmt.Sprintf("name = %s", instanceType)
+		instanceTypes, err := client.GetMachineTypes(context.TODO(), ic.GCP.ProjectID, filter)
+		if err != nil {
+			return append(allErrs, field.InternalError(fieldPath, err))
+		}
+		if typeMeta, ok := instanceTypes[instanceType]; ok {
+			if typeMeta.GuestCpus < req.minimumVCpus {
+				errMsg := fmt.Sprintf("instance type does not meet minimum resource requirements of %d vCPUs", req.minimumVCpus)
+				allErrs = append(allErrs, field.Invalid(fieldPath.Child("instancetype"), instanceType, errMsg))
+			}
+			if typeMeta.MemoryMb < req.minimumMemory {
+				errMsg := fmt.Sprintf("instance type does not meet minimum resource requirements of %d MiB Memory", req.minimumMemory)
```

Comment:
Will this lead to any confusion for the user if we are reporting the memory requirements as MiB whereas GCP reports it as MB? I don't think that I often see MiB used in the context of memory.

Reply:
Probably so, was brought over from AWS code doing the same. I reworded to MB for clarity. https://github.com/openshift/installer/pull/4329#event-3949892939